cloudigrade/account/reports/__init__.py

```diff
@@ -179,3 +187,82 @@ def _calculate_instance_usage(start, end, events):
         time_running += diff.total_seconds()
 
     return time_running
+
+def validate_event(event, start):
+    """
+    Ensure that the event is relevant to our time frame.
+
+    Args:
+        event: (InstanceEvent): The event object to evaluate
+        start (datetime.datetime): Start time (inclusive)
+
+    Returns:
+        bool: A boolean regarding whether or not we should inspect the event further.
+    """
+    valid_event = True
+    # if the event occurred outside of our specified period (ie. before start) we should
+    # only inspect it if it was a power on event
+    if event.occurred_at < start:
+        if event.event_type == InstanceEvent.TYPE.power_off:
+            valid_event = False
+    return valid_event
+
+def get_account_overview(account, start, end):
+    """
+    Generate an overview of an account over a specified amount of time.
+
+    Args:
+        start (datetime.datetime): Start time (inclusive)
+        end (datetime.datetime): End time (exclusive)
+        account (AwsAccount): AwsAccount object
+
+    Returns:
+        dict: An overview of the instances/images/rhel & openshift images for the specified
+        account during the specified time period."""
+    instances = []
+    images = []
+    rhel = []
+    openshift = []
+    # if the account was created right at or after the end time, we cannot give meaningful
+    # data about the instances/images seen during the period, therefore we need to make sure
+    # that we return None for those values
+    if end <= account.created_at:
+        logger.info(_('Account "{0}" was created after "{1}", therefore there is no data on '
+                      'its images/instances during the specified start and end dates.').format(
+            account,
+            end))
+        total_images, total_instances, total_rhel, total_openshift = None, None, None, None
+    else:
+        cloud_helper = helper.get_report_helper(AWS_PROVIDER_STRING, account.aws_account_id)
+        # _get_relevant_events will return the events in between the start & end times & if
+        # no events are present during this period, it will return the last event that occurred
+        events = _get_relevant_events(start, end, cloud_helper)
+        for event in events:
+            valid_event = validate_event(event, start)
+            if valid_event:
+                instances.append(event.instance.id)
+                images.append(event.ec2_ami_id)
+                image = AwsMachineImage.objects.get(ec2_ami_id=event.ec2_ami_id)
+                for tag in image.tags.all():
+                    if tag.description == 'rhel':
```

Comment:
It's not something we need to fix _now_ with your commit here, but as I see we are using magic string tokens to indicate "rhel" and "openshift", I keep thinking that we should have a better way of recognizing these special tags. I'm just "thinking aloud" here but recording it so we don't forget…